### PR TITLE
Change icon if path is different

### PR DESCRIPTION
**Pull request by mikrohard**
If we change the icon url in the backend it should be also changed in the frontend.

Why is it even neccessary to check if m_strIconPath is empty? I may be better to drop the if sentence entirely.

### CODE REVIEW

**Comment by opdenkamp** [comment]
because people can choose to set another icon in xbmc. this actually needs a state boolean that is set when the user changes the icon to something else, so the backend doesn't reset the icon.

**Comment by mikrohard** [comment]
Ok... I get it. So we need to extend the channel db with a state bool, which is set when the user changes the icon. And then we allow backend icon updates only if this bool is in a specific state. I'll update this later...

**Comment by opdenkamp** [comment]
great

**Comment by mikrohard** [comment]
Ok... this works for me. This also fixes the bug where the user could not remove the custom icon with the none option. Now if the user choses "None" the state bool is reset and the backend icon is used again (or none if there is no backend icon).

File: xbmc/pvr/PVRDatabase.cpp [hunk 1]
```diff
@@ -255,6 +256,8 @@ bool CPVRDatabase::UpdateOldVersion(int iVersion)
           database.Close();
         }
       }
+      if (iVersion < 20)
+        m_pDS->exec("ALTER TABLE channels ADD bIsUserSetIcon bool;");
```
Comment (opdenkamp) [inline xbmc/pvr/PVRDatabase.cpp]: please remove the ; at the end. I've heard that the mysql wrapper doesn't understand the query if it is present at the end. I'll have to remove the other leftovers too I see.

File: xbmc/pvr/PVRDatabase.cpp [hunk 2]
```diff
@@ -328,11 +331,11 @@ int CPVRDatabase::Persist(const CPVRChannel &channel, bool bQueueWrite /* = fals
   {
     /* update channel */
     strQuery = FormatSQL("REPLACE INTO channels ("
-        "iUniqueId, bIsRadio, bIsHidden, "
+        "iUniqueId, bIsRadio, bIsHidden, bIsUserSetIcon, "
         "sIconPath, sChannelName, bIsVirtual, bEPGEnabled, sEPGScraper, iLastWatched, iClientId, "
         "iClientChannelNumber, sInputFormat, sStreamURL, iEncryptionSystem, idChannel, idEpg) "
-        "VALUES (%i, %i, %i, '%s', '%s', %i, %i, '%s', %u, %i, %i, '%s', '%s', %i, %i, %i);",
-        channel.UniqueID(), (channel.IsRadio() ? 1 :0), (channel.IsHidden() ? 1 : 0),
+        "VALUES (%i, %i, %i, %i, '%s', '%s', %i, %i, '%s', %u, %i, %i, '%s', '%s', %i, %i, %i);",
```
Comment (opdenkamp) [inline xbmc/pvr/PVRDatabase.cpp]: same here (yeah, I know it was already there in the first place)

File: xbmc/pvr/windows/GUIWindowPVRChannels.cpp [hunk 3]
```diff
@@ -448,7 +448,7 @@ bool CGUIWindowPVRChannels::OnContextButtonSetThumb(CFileItem *item, CONTEXT_BUT
       if (strThumb == "thumb://None")
         strThumb = "";
 
-      channel->SetIconPath(strThumb, true);
+      channel->SetIconPath(strThumb, true, true);
```
Comment (opdenkamp) [inline xbmc/pvr/windows/GUIWindowPVRChannels.cpp]: you missed a couple of other calls with this. e.g. CGUIDialogPVRChannelManager::PersistChannel()
better idea to do it the other way around: default the boolean to true, and set it to false when loading the channel info from the client

File: xbmc/pvr/PVRDatabase.cpp (hunk 2 above)
Comment (mikrohard) [inline xbmc/pvr/PVRDatabase.cpp]: Will do... no problem.

File: xbmc/pvr/windows/GUIWindowPVRChannels.cpp (hunk 3 above)
Comment (mikrohard) [inline xbmc/pvr/windows/GUIWindowPVRChannels.cpp]: I see... the icon can also be changed with the channel manager... I missed that. 

What about the SetIconPath calls in CPVRChannelGroup::SearchAndSetChannelIcons? Should those icons be updated from the backend or not?

**Comment by mikrohard** [comment]
I changed the default bool to true... but I'm still not sure about CPVRChannelGroup::SearchAndSetChannelIcons calls.

**Comment by opdenkamp** [comment]
merged, thanks

**Pull request merged**